core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -555,12 +555,15 @@ private[spark] class BlockManager(
 
   /**
    * Return a list of locations for the given block, prioritizing the local machine since
-   * multiple block managers can share the same host.
+   * multiple block managers can share the same host, then try to get the same rack data.
```

Comment:
How about "then try to get the same rack data" -> "followed by hosts on the same rack" ?